src/status_im/contexts/wallet/wallet_connect/responding_events.cljs

```diff
@@ -0,0 +1,96 @@
+(ns status-im.contexts.wallet.wallet-connect.responding-events
+  (:require [re-frame.core :as rf]
+            [status-im.constants :as constants]
+            [status-im.contexts.wallet.wallet-connect.core :as wallet-connect-core]
+            [taoensso.timbre :as log]))
+
+(rf/reg-event-fx
+ :wallet-connect/respond-current-session
```

Comment:
I tried to avoid creating abstractions at this point and made it as explicit as I could, with an event for (almost) each rpc method (personal_sign, sign_typed, eth_sign, etc,) for processing and responding. The reason for this is because it feels a bit early to join them, because we don't have all the methods there just yet. Some will be added soon (sign_transaction, send_transaction), others later (permissions, etc.) and they will have different behaviour in some cases. I think after we add the transaction parts, it might get clearer where we can simplify the events. WDYT?

Reply:
> This is the kind of event that I think should be an effect instead.

In an abstract sense yes (according to your explanation), but its actually closer to an event than an effect. 

I have something similar in my PR, and I just wrote a util function. 

Doesn't feel like an effect because there are no *side-effects*.

Reply:
> Doesn't feel like an effect because there are no side-effects.

Which part exactly doesn't feel like an effect?

Reply:
If I had the time today I'd attempt to refactor and compare. My intuition here just tells me that we're over-using events and effects when we could have even a separate namespace isolating wallet connect operations that is independent of re-frame. Then the re-frame effects/events would be very shallow shells and we would up with less of them (generally a good thing). Every refactor suggestion is theoretically better, but only when we attempt we know better. So there's a chance I am the one over-complicating :)

Anyway, food for thought :)